pkg/syncer/metadatasyncer.go

```diff
@@ -781,10 +799,24 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 				}
 				metadataList = append(metadataList, cnstypes.BaseCnsEntityMetadata(podMetadata))
 				containerCluster := cnsvsphere.GetContainerCluster(metadataSyncer.configInfo.Cfg.Global.ClusterID, metadataSyncer.configInfo.Cfg.VirtualCenter[metadataSyncer.host].User, metadataSyncer.clusterFlavor)
-
+				var volumeHandle string
+				var err error
+				if metadataSyncer.configInfo.Cfg.FeatureStates.CSIMigration && pv.Spec.VsphereVolume != nil {
+					volumeHandle, err = GetVolumeID(ctx, pv)
+					if err != nil {
+						log.Errorf("failed to get volume id for volume name: %q with err=%v", pv.Name, err)
+						return
+					}
+				} else {
+					volumeHandle = pv.Spec.CSI.VolumeHandle
```

Comment:
Here we are assuming POD uses only CSI PV and in-tree migrated PV, but POD can use other types of PVs for example nfs, empty dir as well in the spec.

Reply:
Before this change, we never handled such cases. Is that needed ?

Reply:
we do check this for CSI PV - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/util.go#L59-L82

Reply:
this is getting called earler in the code. Ignore this.